pydatastructs/linear_data_structures/algorithms.py

```diff
@@ -620,6 +621,95 @@ def swap(i, j):
                 swapping = False
         lower = lower + 1
 
+    return array
+
+def quick_sort(array: Array, **kwargs) -> Array:
+    """
+    Performs quick sort on the given array.
+
+    Parameters
+    ==========
+
+    array: Array
+        The array which is to be sorted.
+    start: int
+        The starting index of the portion
+        which is to be sorted.
+        Optional, by default 0
+    end: int
+        The ending index of the portion which
+        is to be sorted.
+        Optional, by default the index
+        of the last position filled.
+    comp: lambda/function
+        The comparator which is to be used
+        for sorting. If the function returns
+        False then only swapping is performed.
+        Optional, by default, less than or
+        equal to is used for comparing two
+        values.
+
+    Returns
+    =======
+
+    output: Array
+        The sorted array.
+
+    Examples
+    ========
+
+    >>> from pydatastructs import OneDimensionalArray as ODA, quick_sort
+    >>> arr = ODA(int, [5, 78, 1, 0])
+    >>> out = quick_sort(arr)
+    >>> str(out)
+    '[0, 1, 5, 78]'
+    >>> arr = ODA(int, [21, 37, 5])
+    >>> out = quick_sort(arr)
+    >>> str(out)
+    '[5, 21, 37]'
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Quicksort
+    """
+    comp = kwargs.get("comp", lambda u, v: u <= v)
+
+    def partition(low, high):
+        i = (low - 1)
+        x = array[high]
+        for j in range(low , high):
+            if _comp(array[j], x, comp) is True:
+                i = i + 1
+                array[i], array[j] = array[j], array[i]
+        array[i + 1], array[high] = array[high], array[i + 1]
+        return (i + 1)
+
+    lower = kwargs.get('start', 0)
+    upper = kwargs.get('end', len(array) - 1)
+    stack, top = [0] * (upper - lower + 1), -1
```

Comment:
There is a `Stack` class (see below).
https://github.com/codezonediitj/pydatastructs/blob/78d8254156e6984ff556eb16c468cb53ab132a54/pydatastructs/miscellaneous_data_structures/stack.py#L9

Try using the above as stack. 

P.S.  The iterative version seems nice.

Reply:
Cool I will use the stack there in the program ^ ^